frontend/packages/dev-console/src/components/import/validation-schema.ts

```diff
@@ -36,7 +36,10 @@ export const applicationNameValidationSchema = yup.object().shape({
   selectedKey: yup.string(),
   name: yup.string().when('selectedKey', {
     is: CREATE_APPLICATION_KEY,
-    then: yup.string().required('Required'),
+    then: yup
+      .string()
+      .max(63, 'Cannot be longer than 63 characters.')
+      .required('Required'),
```

Comment:
@jerolimov This would only work when we already have an application and CREATE_APPLICATION_KEY is selected from the dropdown. For this to  work properly please remove `when` block validation and simply add `max`
```
name: yup.string().max(63, 'Cannot be longer than 63 characters.').required('Required'),
```
Currently, this is not working if there are no applications in the project 
![Screenshot from 2020-05-20 22-41-27](https://user-images.githubusercontent.com/9278015/82476080-22334980-9aeb-11ea-9c84-0f7e4fe21f22.png)

Reply:
@sahil143 You're right, adding the max check to the when condition was wrong, because it ignores the case that there is "no application available, yet".

But removing the when condition would remove the required check when there is already an application available. I think this time this matches all cases

* No app is available:
  * Show a text input field. Input is not required, but max length is validated.
* An app is available:
  * Select an existing app works
  * If the user selected create new app -> Input is required (max length also)
  * Unassigned -> Text field was not shown, so input is not required.

Reply:
Agreed! This will work properly.